markdown/developers/adding_modules.md

```diff
@@ -523,6 +523,84 @@ If the module absolute cannot run using tiny test data, there is a possibility t
 nextflow run tests/modules/<nameofmodule> -entry test_<nameofmodule> -c tests/config/nextflow.config -stub-run
 ```
 
+## Into the `meta` map
+
+As you've noticed most modules take an input of
+
+```nextflow
+input:
+tuple val(meta), path(reads)
+```
+
+You have probably also noticed, this pattern doesn't work out of the box with [fromFilePairs](https://www.nextflow.io/docs/edge/channel.html#fromfilepairs)
+
+The difference between the two:
+
+```nextflow
+// fromFilePairs
+filepairs = [SRR493366, [/my/data/SRR493366_1.fastq, /my/data/SRR493366_2.fastq]]
+
+// meta map
+meta_map = [[id: 'test', single_end: false], // meta map
+            [/my/data/SRR493366_1.fastq, /my/data/SRR493366_2.fastq]]
+```
+
+As you can see the difference, they are both [groovy lists](https://www.tutorialspoint.com/groovy/groovy_lists.htm).
+However, the filepairs just has a `val` that is a string, where as the `meta_map` the first value in the list, is a [groovy map](https://www.tutorialspoint.com/groovy/groovy_maps.htm), which is like a python dictionary.
+The only required value is `id` for most of the modules, however they occasionally need things like `single_end`.
+
+### Common patterns
```

Comment:
```suggestion
### Generating the `meta map`
```

Reply:
Going to leave this, because I'm trying to collect some common nf-core nextflow patterns here. I don't think this is their final place, but it's better to collect them.